2016-2017_Starstruck/Module Prototypes/src/LCD.c

```diff
@@ -1,36 +1,125 @@
 #include <main.h>
+#include <string.h>
 
-MenuItem *lcdmInit() {
-  char title[] = "<    Select    >";
-  MenuItem root = {.name = title,
-                   .previous = malloc(sizeof(MenuItem *)),
-                   .next = malloc(sizeof(MenuItem *)),
-                   .select = malloc(sizeof(MenuItem *)),
-                   .action = malloc(sizeof(Function))
-                  };
-  return &root;
+MenuItem *current;
+FILE *port;
+
+void pulseMotor(unsigned char port);
+
+MenuItem *lcdmInit(FILE *lcdPort) {
+  port = lcdPort;

```

Comment:
yea, port needs to be stored locally for use outside of init (setting lcd text in lcdLoop)